### PR TITLE
add TestConfChangeRemoveLeader3B

### PR DESCRIPTION
**Pull request by ZYFZYF**
Added a simple test to check if the request to remove the leader in a membership change is handled correctly, as the author said in the paper, especially when there are only two members, interesting things may happen when removing the leader

### CODE REVIEW

File: kv/test_raftstore/test_test.go [hunk 1]
```diff
@@ -597,6 +597,47 @@ func TestBasicConfChange3B(t *testing.T) {
...
+	cluster.MustRemovePeer(1, NewPeer(1, 1))
+	cluster.MustPut([]byte("k0"), []byte("v0"))
+	MustGetNone(cluster.engines[1], []byte("k0"))
+
+	// rejoin and become leader, now store 1 can see it
+	cluster.MustAddPeer(1, NewPeer(1, 1))
+	cluster.MustTransferLeader(1, NewPeer(1, 1))
+	cluster.MustPut([]byte("k1"), []byte("v1"))
+	// wait for apply
+	time.Sleep(500 * time.Millisecond)
```
Comment (Connor1996) [inline kv/test_raftstore/test_test.go]: `MustGetEqual` does the waiting

Comment (ZYFZYF) [reply kv/test_raftstore/test_test.go]: Removed, thx

**Review by Connor1996 — COMMENTED** [review]
new test case may affect the grading script, may need adjust the script PTAL @yanguwan

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by kikimo** [comment]
@ZYFZYF Removing peer in a two node cluster is a very interesting problem when using an apply after commit conf change strategy. What I am interesting is that, can you raft implementtation still pass this test(or other 3B test cases) when running this these cases repeatly for 512 times or more.

**Comment by ZYFZYF** [comment]
> @ZYFZYF Removing peer in a two node cluster is a very interesting problem when using an apply after commit conf change strategy. What I am interesting is that, can you raft implementtation still pass this test(or other 3B test cases) when running this these cases repeatly for 512 times or more.

I ran this test 512 times and found no problems. As for other tests, it takes a long time to run 512 times(~100+h). I am still working on it(have found some bugs and fixed)

**Pull request merged**